### DISCUSSION TITLE
Question about assembling APK, DexIndexOverFlowException.

### DISCUSSION THREAD

**Discussion by Pancax** [osmandapp/OsmAnd#12029] [Q&A]
Sorry, if I have placed this in the wrong place.

I am trying to assemble a debug apk for this project (I don't want the tests to run), but every time I try 
```
gradle clean assembleFullLegacyFatDebug --stacktrace -x test
```
I get a runtimeexceeption with Dex saying it was unable to merge, 
```
com.android.dex.DexIndexOverflowException: Cannot merge new index 72450 into a non-jumbo instruction.
```

I am unfamiliar with dex and have tried to use some suggested google answers, but I figured ya'll might know what is going on.
I have added 
```
dex.force.jumbo=true 
```
in my gradle.properties file and in the appropriate build.gradle files I added 
```
dexOptions{
   jumboMode true
}
```
and I still run into this issue.
Any help would be appreciated :)

**Comment by vshcherb** [comment]
Probably you need to update android sdk tools. What you mentionned should work correctly and peforms well on server side.

**Reply by Pancax** [reply]
Thanks for your answer, sorry I am replying to you so late. I updated my android studio sdk tools and I still run into this error while building. Do you have any other guesses as to what might be going on? Thanks.

**Comment by poretsky** [comment]
Just a suggestion. Make sure you use the java compiler provided with the
Android studio.
